src/librustc_borrowck/borrowck/mod.rs

```diff
@@ -236,6 +296,47 @@ impl<'tcx> Loan<'tcx> {
     pub fn loan_path(&self) -> Rc<LoanPath<'tcx>> {
         self.loan_path.clone()
     }
+
+    pub fn kind(&self) -> ty::BorrowKind {
+        self.kind
+    }
+
+    pub fn span(&self) -> Span {
+        self.span
+    }
+
+    pub fn span_in_def(&self, def_id: DefId, tcx: TyCtxt) -> Option<Span> {
```

Comment:
Definitely a comment would be good -- with some examples. I would think `span_of_def` or `span_for_def` would be a better name, to start. Is this the right place for this helper, anyway?